src/engine/BoundManager.cpp

```diff
@@ -296,3 +304,213 @@ void BoundManager::registerRowBoundTightener( IRowBoundTightener *ptrRowBoundTig
     ASSERT( _rowBoundTightener == nullptr );
     _rowBoundTightener = ptrRowBoundTightener;
 }
+
+void BoundManager::explainBound( unsigned variable, bool isUpper, Vector<double> &explanation ) const
+{
+     ASSERT( shouldProduceProofs() && variable < _size );
+    auto temp = _boundExplainer->getExplanation( variable, isUpper );
+    explanation = Vector<double>( temp.size() );
+    for ( unsigned i = 0; i < temp.size(); ++i )
+        explanation[i] = *temp[i];
+}
+
+bool BoundManager::tightenLowerBound( unsigned variable, double value, const TableauRow &row )
+{
+    bool tightened = setLowerBound( variable, value );
+
+    if ( tightened )
+    {
+        if ( shouldProduceProofs() )
+            _boundExplainer->updateBoundExplanation( row, false, variable );
```

Comment:
This "false" parameter is not very informative. Can we change it into an enum? Or at least, pass a named boolean variable here